presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```
@@ -253,6 +257,18 @@ public ConnectorSplitSource getSplits(
         return splitSource;
     }
 
+    private static Optional<Domain> getPathDomain(TupleDomain<? extends ColumnHandle> domainPredicate)
+    {
+        if (domainPredicate.isNone()) {
```

Can you throw an exception since none is not valid input at this point?